applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py

```diff
@@ -373,6 +373,7 @@ def Initialize(self):
 
         # Set the distance modification process
         self.__GetDistanceModificationProcess().ExecuteInitialize()
+        self.__GetDistanceModificationProcess().ExecuteInitializeSolutionStep()
```

Comment:
could you add a comment to why this is necessary? Otherwise it confuses the reader as to why `ExecuteInitializeSolutionStep` is being called in `Initialize`

or even more, shouldn't this be integrated into the `DistanceModificationProcess`?

Reply:
Well, we can call the `ExecuteBeforeSolutionLoop` of the `DistanceModificationProcess`, which does actually the same.

Reply:
hm no clue in the end what is happening, just saying from an outside perspective this is confusing :)

Reply:
In a common CFD case, what we normally do is to call the distance computation processes before the solver `Initialize()` call. Now imagine that the level-set you get has any ill-conditioned cut.

Before this PR, the first time you correct the distance is in the `InitializeSolutionStep()`. This means that if there's any process that requires the level-set at some point between the `Initialize()` and the `InitializeSolutionStep()` (e.g. `ExecuteBeforeSolutionLoop()` of any process) it will crash. This aims to fix this.